homeassistant/components/airthings_ble/__init__.py

```diff
@@ -42,6 +43,8 @@ async def _async_update_method():
         ble_device = bluetooth.async_ble_device_from_address(hass, address)
         airthings = AirthingsBluetoothDeviceData(_LOGGER, elevation, is_metric)
 
+        await close_stale_connections(ble_device)
```

Comment:
This method is only intended to be used when the integration is setup to clean up if things didn't get shutdown cleanly. The library should implement proper connection management with disconnected callbacks so it shouldn't be needed in the update path.

It's completely fine to move it above in the setup process but if the intent is to workaround a flaw in the library, the library should be fixed instead

Reply:
I just realized there is a problem with close stale connections not actually being able to close the connections if the device stops advertising when its connected like some switchbot devices do.  I'm not sure if that applies to airthings-ble though.

Regardless the solution is in https://github.com/home-assistant/core/pull/106835

If you aren't sure which one to use `close_stale_connections_by_address` is the better choice